### DISCUSSION TITLE
Feature Request: Add Advanced Filters and Rules for Trailer Downloads

### DISCUSSION THREAD

**Discussion by alexkahler** [nandyalu/trailarr#51] [Ideas]
One improvement I'd love to see in Trailarr is the ability to set up advanced filters for trailer downloads. Specifically, it would be helpful to have more control over which trailers are downloaded based on various criteria like release status and library availability.

For instance:

I want Trailarr to only download trailers for movies that are unreleased or currently in cinemas.
Alternatively, I'd like the option to restrict downloads to trailers for movies that are already released and present in my library, avoiding trailers for unreleased or unavailable content.
Additionally, it would be powerful if these filters could act as actionable rules. For example, users could set Trailarr to automatically download trailers for all items where specified filters apply. This functionality could extend to managing monitoring status as well, based on factors such as:

Whether a movie/show is already in my library.
If the content has just been added.
The monitoring status in Radarr/Sonarr.
The status of the trailer (e.g., downloaded or missing).
This level of customization would allow users to tailor trailer downloads and monitoring status to match their specific needs, ensuring Trailarr only downloads trailers that are most relevant and in line with their preferences.

**Comment by nandyalu** [comment]
This has been implemented in `v0.4.0` with `Profiles`.

It doesn't have many options at the moment. Let me know what other options you would like to have and I'll try to add them!
